tests/unit/test_dwave_sampler.py

```diff
@@ -127,6 +126,33 @@ def setUp(self, MockClient):
 
         self.sampler.solver = MockSolver()
 
+    def test_solver_init(self):
+        # test deprecation warning raised for `solver_features`, but that it still works
+
+        # assertWarns not available in py2
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter("always")
+            DWaveSampler(solver_features={'qpu': True})
+            self.assertEqual(len(w), 1)
+            self.assertTrue(issubclass(w[-1].category, DeprecationWarning))
+
+        sampler = DWaveSampler(solver_features={'qpu': True})
```

Comment:
If this is accessing an online system is should go under integration tests.

Reply:
Yup. Reworked to work in offline mode.